### DISCUSSION TITLE
Sending negative Coordinates

### DISCUSSION THREAD

**Discussion by tom-p-zh** [winder/Universal-G-Code-Sender#2031] [Help]
(quite experienced cnc-user - new to grbl / ugs)
maybe i missed some settings. after setting up an experimental pencil-plotter - everything works fine, but i am not able to send negative coordinates from a g-code file. no homing, no soft ends set.
A really simple
`G0X-10`
is send from the text window as (and also feedbacked in the command like this)
`G0X10`
(and the machine moves +10)
from the UGS-command-line
`G0X-10`
works fine and the machine moves as expected.
is it a bug ? some special gcode syntax needed ? do i miss some settings to allow working in quadrant 2,3,4 ?
thanks for support - kind regards from zürich / switzerland - tom

**Comment by breiler** [comment]
Which version of UGS are you running? 
Can we have a sample gcode file where this happening?

There was a problem some time ago where minus characters were typed using a two byte UTF-8 character instead of a ASCII-character which couldn't be understood by the controller.

**Reply by tom-p-zh** [reply]
Dear Joacim - thanks for having a look at it. 
after restarting the macbook, UGS, the arduino now everything works as expected.
maybe it was my fault not handling changes to the file / saving correctly ... i'll play around now and see if this issue happens again.

the minus in the file is a hex 2D
checked with an ascii editior - i think this is fine.


```
Product Version: Universal Gcode Platform 20220810
Java: 13.0.1; OpenJDK 64-Bit Server VM 13.0.1+9
Runtime: OpenJDK Runtime Environment 13.0.1+9
System: Mac OS X version 10.15.7 running on x86_64; UTF-8; en_US (ugsplatform)

```

**Reply by breiler** [reply]
Excellent, good luck! 😃
